pylint_django/transforms/foreignkey.py

```diff
@@ -16,6 +16,15 @@ def is_foreignkey_in_class(node):
     if not isinstance(node.parent.parent, ClassDef):
         return False
 
+    is_in_django_model_class = False
+    # Make sure the class is the subclass of django.db.models.Model
+    for ancestor in node.parent.parent.ancestors():
+        if ancestor.name == 'Model':
+            is_in_django_model_class = True
+            break
```

Comment:
See `node_is_subclass()` in utils.py. If that works it should be easier for you.

Reply:
Thanks, Alexander, updated.